db/mongo/mongoClient.go

```diff
@@ -0,0 +1,106 @@
+package mongo
+
+import (
+	"context"
+	"fmt"
+
+	guuid "github.com/google/uuid"
+	"github.com/pkg/errors"
+	"github.com/spf13/viper"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+
+	"github.com/abdulmoeid7112/student-backend-portal/config"
+	"github.com/abdulmoeid7112/student-backend-portal/db"
+	"github.com/abdulmoeid7112/student-backend-portal/models"
+)
+
+const (
+	collectionName = "Student"
+)
+
+func init() {
+	db.Register("mongo", NewMongoClient)
+}
+
+type client struct {
+	conn *mongo.Client
+}
+
+func (m client) AddStudent(student *models.Student) (string, error) {
+	if student.ID == "" {
+		student.ID = guuid.New().String()
+		collection := m.conn.Database(viper.GetString(config.DbName)).Collection(collectionName)
+		if _, err := collection.InsertOne(context.TODO(), student); err != nil {
+			return "", errors.Wrap(err, "failed to add student")
+		}
+	}
+	return student.ID, nil
+}
+
+func (m client) UpdateStudent(student *models.Student) error {
+	collection := m.conn.Database(viper.GetString(config.DbName)).Collection(collectionName)
+	if _, err := collection.UpdateOne(context.TODO(), bson.M{"_id": student.ID}, bson.M{"$set": student}); err != nil {
+		return errors.Wrap(err, "failed to update student")
+	}
+	return nil
+}
+
+func (m client) GetStudentByID(id string) (*models.Student, error) {
+	var stu *models.Student
+	collection := m.conn.Database(viper.GetString(config.DbName)).Collection(collectionName)
+	if err := collection.FindOne(context.TODO(), bson.M{"_id": id}).Decode(&stu); err != nil {
+		if mongo.ErrNoDocuments != nil {
+			return nil, errors.Wrap(err, "failed to fetch student....not found")
+		}
+		return nil, err
+	}
+	return stu, nil
+}
+
+func (m client) ListStudent(filter map[string]interface{}, lim int, off int) ([]*models.Student, error) {
+	var std []*models.Student
+	fields2 := bson.M{}
+	for key, val := range filter {
+		fields2[fmt.Sprint(key)] = val
+	}
+	options := options.Find()
+	options.SetLimit(int64(lim))
+	options.SetSkip(int64(off))
```

Comment:
try to use options inline.

Reply:
I didn't find any suitable solution for this except this one.

Reply:
Fixed